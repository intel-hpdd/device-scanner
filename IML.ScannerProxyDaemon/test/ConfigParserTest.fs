// Copyright (c) 2018 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.ScannerProxyDaemon.ConfigParserTest

open Fable.Import.Jest
open Matchers

open ConfigParser

testList "Data Handler" [
  Test("Should filter out all but the base64 encoded string of 'server'", fun () ->
    [| "foo"; "bar"; "c2VydmVy" |]
      |> (filterFileName "server")
      |> Seq.toArray
      |> toMatchSnapshot
  )

  Test("Should parse url from map", fun () ->
    Map.empty
      |> Map.add "url" "https://foo.com:443/agent/"
      |> parseUrl
      |> toMatchSnapshot
  )
]
