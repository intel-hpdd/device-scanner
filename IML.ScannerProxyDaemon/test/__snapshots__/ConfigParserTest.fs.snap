// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Handler Should filter out all but the base64 encoded string of 'server' 1`] = `
Array [
  "c2VydmVy",
]
`;

exports[`Data Handler Should parse url from map 1`] = `"foo.com"`;
